clio-client/src/main/scala/org/broadinstitute/clio/client/commands/ClioParsers.scala

```diff
@@ -0,0 +1,59 @@
+package org.broadinstitute.clio.client.commands
+
+import akka.http.scaladsl.model.headers.OAuth2BearerToken
+import caseapp.core.ArgParser
+import enumeratum.{Enum, EnumEntry}
+
+import scala.reflect.ClassTag
+import scala.util.{Failure, Success, Try}
+
+import java.time.OffsetDateTime
+import java.util.UUID
+
+/**
+  * Argument parsers to use in building command parsers.
+  *
+  * We use caseapp to auto-magically build command-line parsers for
+  * our data types. The docs are sparse, but for what is there see:
+  *
+  * https://github.com/alexarchambault/case-app/blob/master/README.md
+  */
+trait ClioParsers {
+  implicit def enumEntryParser[T <: EnumEntry: Enum](
+    implicit c: ClassTag[T]
+  ): ArgParser[T] = {
+    ArgParser.instance[T]("entry") { entry =>
+      val value = implicitly[Enum[T]].withNameOption(entry)
+      Either.cond(
+        value.isDefined,
+        value.get,
+        s"Unknown enum value $entry for type ${c.runtimeClass.getName} "
+      )
+    }
+  }
+
+  implicit val oauthBearerTokenParser: ArgParser[OAuth2BearerToken] = {
+    ArgParser.instance[OAuth2BearerToken]("token") { token =>
+      //no need for left since this should never fail
```

Comment:
Why should this never fail?

Reply:
Because all strings are valid bearer tokens as far as parsing is concerned. (Whether or not Clio accepts the token is a different problem)

Reply:
But what if the constructor to OAuth2BearerToken fails?

Reply:
It won't, it's a case class that takes a string with no validation: https://github.com/akka/akka-http/blob/master/akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpCredentials.scala#L46